magento/pwa-studio
# Pwa 2482

## Description

PWA-2420 asked for out of stock products not to be shown in search results but we decided that was too heavy of a treatment, especially when the admin has set "show out of stock products on the storefront" to YES.

Instead, experiment with other treatments like de-prioritizing the product or adding some sort of "out of stock" indicator on the search results listing.

Acceptance Criteria

Display Out of Stock indicator for a product on search results page when configuration is set in admin to show, and item is out of stock
Out of Stock products are not shown on search results page when configuration is set in admin to not show

## Related Issue







Closes #PWA2482

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps





Test Plan

Given backend is configured to show Out of stock products - Yes

search for out of stock product and verify its indicated as in mocks. Also make sure it is shown in search suggestion.
search for generic string which includes both in and out of stock and verify product display
Reload the browser in above scenario and verify it still works as expected
Verify out of stock indication on search in Chrome, Firefox and safari browser and in mobile device

Given backend is configured to show Out of stock products - No

search for out of stock product and verify it's not displayed both in search suggestion and search page itself.
Test Coverage
Cover changes in unit tests
In case unit test coverage is not enough add cypress mock data test to cover the out of stock display.
#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
